overseer/examples/minimal-example.rs

```diff
@@ -0,0 +1,111 @@
+use std::collections::HashSet;
+use std::time::Duration;
+
+use futures::pending;
+use futures_timer::Delay;
+use kv_log_macro as log;
+
+use overseer::{Overseer, Subsystem, SubsystemContext, SubsystemJob};
+
+struct Subsystem1;
+
+impl Subsystem1 {
+	async fn run(mut ctx: SubsystemContext<usize>)  {
+		loop {
+			match ctx.try_recv().await {
+				Ok(Some(msg)) => {
+					log::info!("Subsystem1 received message {}", msg);
+				}
+				Ok(None) => (),
+				Err(_) => {}
+			}
+
+			Delay::new(Duration::from_secs(1)).await;
+			ctx.send_msg(10).await;
+			pending!();
+		}
+	}
+
+	fn new() -> Self {
+		Self
+	}
+}
+
+impl Subsystem<usize> for Subsystem1 {
+	fn start(&mut self, ctx: SubsystemContext<usize>) -> SubsystemJob {
```

Comment:
not a fan of `start`, `spawn` / `block_on`/`run` tend to give a better idea, of what it does, but that might be a personal pref

Reply:
`block_on` is probably not the best choice since we are not blocking here, but `spawn` works for me